osltweaks.php

```diff
@@ -44,6 +44,26 @@ function osltweaks_civicrm_buildForm($formName, &$form) {
         }
       }
     }
+
+    // If contribution page id exist in $settings['member_redirect_contribution_pages'] array
+    if (array_key_exists($form->_id, ($settings['member_redirect_contribution_pages'] ?? array()))) {
+      // Get current contact id
+      $currentContactId = CRM_Core_Session::singleton()->getLoggedInContactID();
+      // Get memberships count base on the current contact id
+      $membershipCount = civicrm_api3('Membership', 'getcount', [
+        'contact_id' => $currentContactId,
+      ]);
+
+      // Redirect if there are results on memberships api
+      if ($membershipCount) {
+        $redirectFormId = $form->_id;
+        // Parse current URL to get the current q
+        parse_str(html_entity_decode(parse_url($form->controller->_entryURL, PHP_URL_QUERY)), $parseDecodedUrl);
+        // Get the redirect id in $settings['member_redirect_contribution_pages'] base on $redirectFormId
+        $redirect = $settings['member_redirect_contribution_pages'][$redirectFormId];
```

Comment:
Sorry @lucky091588 this is close but not exactly what I was meaning in my previous comment. This variable `$redirect` stores an integer form ID. Therefore it should have a name that indicates that it stores a form ID. Can fix that?

Reply:
@twomice Sorry, I misunderstood the previous comment. See latest commit for the changes.